crates/storage/codecs/src/alloy/header.rs

```diff
@@ -23,7 +23,7 @@ struct Header {
     logs_bloom: Bloom,
     difficulty: U256,
     number: BlockNumber,
-    gas_limit: u64,
```

Comment:
I've just undo this change. I suggest that for now we leave the codec as it is without touching it so we don't risk breaking anything. There are other fields to change first. When everything is done we can come back to it and fix things properly, wdyt?

Reply:
Since everything is converging towards using u64 for these fields, I wonder if it would not be better to do what is described here https://github.com/alloy-rs/alloy/issues/1241, that is to say to convert everything to u64 in alloy. This will reduce a lot of modifications to be made in reth and it would naturally eliminate all the conversions when applying the clippy...

I do not really see any specific cases where the use of a u128 is justified except perhaps very specific devnets or local developments but as a general rule, I would say that switching everything to u64 seems more justified to me. What do you think @joshieDo @mattsse @klkvr ?

Reply:
> if it would not be better to do what is described here https://github.com/alloy-rs/alloy/issues/1241,

yeah I know think that'd be better, especially because eth consensus has these fields as u64 and u64::MAX should be plenty for all things gas